src/ocean/io/select/client/FileSystemEvent.d

```diff
@@ -153,6 +208,21 @@ class FileSystemEvent : ISelectClient
 
     /***********************************************************************
 
+        Replace the handle delegate
+
+        Params:
+            handler = event handler
+
+    ***********************************************************************/
+
+    public void setHandler ( DirHandler handler )
```

Comment:
This is something I hoped I'll get the input on. Yes, all cases covered by `DirHandler` are covered by `Handler`. 

Initially I opted in keeping both, because  `DirHandler` contains one argument that's usable only in context of monitoring the directory. I felt user would get this dummy argument which is useless for most of the usage scenarios.

Ideally, I would like to rename `DirHandler` to `HandlerEx`, deprecate `Handler` and remove in the next major. In the meantime, based on the assumption, that this class is just a dummy wrapper around `inotify` and `ISelectClient`, expecting user to know internals of `inotify`, mechanism, I would create utility modules which covers the 99% of the usage - `FileCreateModifyEvent` and `FileModifyEvent` (for example) which would internally use `FileSystemEvent`.